demisto_sdk/commands/changelog/changelog.py

```diff
@@ -0,0 +1,214 @@
+from demisto_sdk.commands.common.tools import get_yaml
+
+CHANGELOG_FOLDER = Path(f"{git_path()}/.changelog")
+CHANGELOG_MD_FILE = Path(f"{git_path()}/CHANGELOG.md")
+RELEASE_VERSION_REGEX = re.compile(r"v\d{1,2}\.\d{1,2}\.\d{1,2}")
+
+yaml = DEFAULT_YAML_HANDLER
+
+
+class Changelog:
+    def __init__(
+        self, pr_number: str, pr_name: str = "", release_version: str = None
+    ) -> None:
+        self.pr_number = pr_number
+        self.pr_name = pr_name
+        self.handle_error = ""
+
+    """ VALIDATE """
+
+    def validate(self) -> None:
+        """
+        ...
+        """
+        if self.is_release():
+            self._validate_release()
+        else:
+            self._validate_branch()
+
+    """ INIT """
+
+    def init(self) -> None:
+        """
+        Creates a new log file for the current PR with initial values,
+        then the user has to change the values manually
+        """
+        if self.is_release():
+            logger.error(
+                "This PR is for release, please use in changelog release command"
+            )
+
+        log = INITIAL_LOG
+        log["pr_number"] = self.pr_number
+
+        with (CHANGELOG_FOLDER / f"{self.pr_number}.yml").open("w") as f:
+            yaml.dump(log, f)
+
+        logger.info(
+            f"The creation of the log file .changelog/{self.pr_number}.yml is complete,\n"
+            "Go to the file and edit the initial values."
+        )
+
+    """ RELEASE """
+
+    def release(self) -> None:
+        if not self.is_release():
+            raise ValueError(
+                "The name of the PR is not correct for a release,\n"
+                "The name of the PR should be like the following format: `v1.0.0`"
+            )
+        logs = self.get_all_logs()
+        self.extract_and_build_changelogs(logs)
+        self.cleaning_changelogs_folder()
+        logger.info("The build of the changelog for the release is complete")
+
+    """ HELPER FUNCTIONS """
+
+    def is_changelog_changed(self) -> bool:
+        return (
+            "CHANGELOG.md"
+            in Repo(".").git.diff("HEAD..origin/master", name_only=True).split()
+        )
+
+    def is_log_folder_empty(self) -> bool:
+        return not any(CHANGELOG_FOLDER.iterdir())
+
+    def is_log_yml_exist(self) -> bool:
+        return (CHANGELOG_FOLDER / f"{self.pr_number}.yml").is_file()
+
+    def validate_log_yml(self) -> None:
+        data = get_yaml(CHANGELOG_FOLDER / f"{self.pr_number}.yml")
+
+        try:
+            LogObject(**data)
+        except ValidationError as e:
+            raise ValueError(e.json())
+
+    def get_all_logs(self) -> List[LogObject]:
+        changelogs: List[LogObject] = []
+        for path in CHANGELOG_FOLDER.iterdir():
+            changelog_data = get_yaml(path)
+            try:
+                changelogs.append(LogObject(**changelog_data))
+            except ValidationError as e:
+                self.handle_error += f"{path}: {e.json()}\n"
```

Comment:
let's 
1. change its name to `errors`
2. change its type to `dict[Path, dict]`
3. remove from the class unless used elsewhere